### PR TITLE
HTTP API Cleanup for Member struct

### PR DESCRIPTION
**Pull request by skippi**
Similar to the other PRs. Name changes, typespec cleanups, struct updates, etc.

### CODE REVIEW

**Comment by skippi** [comment]
Think this PR is ready for code review. Was planning to get this done tomorrow, but looks like I got it done earlier than expected.

File: lib/nostrum/api.ex [hunk 1]
```diff
@@ -1229,89 +1229,163 @@ defmodule Nostrum.Api do
...
-          limit: 1..1000,
-          after: integer
-        }) :: error | {:ok, [Nostrum.Struct.Guild.Member.t()]}
-  def get_guild_members(guild_id, options) do
+  If successful, returns `{:ok, members}`. Otherwise, returns a `t:Nostrum.Api.error/0`.
+
+  ## Options
+
+    * `:limit` (integer) - max number of members to return (1-1000) (default: 1)
+    * `:after` (`t:Nostrum.Struct.Snowflake.t/0`) - the highest user id in the previous page (default: 0)
```
Comment (Kraigie) [inline lib/nostrum/api.ex]: I saw you switched this from `User.id()` to `Snowflake.t`, what was the reasoning behind this?

Comment (skippi) [reply lib/nostrum/api.ex]: I was experimenting with the types a little bit to see which one would be more informative to our users.  Looks like I accidentally sent this change with my self code-review. I've fixed it with the recent commit.

**Comment by Kraigie** [comment]
Thanks!

**Pull request merged**